quaistats/quaistats.go

```diff
@@ -675,31 +674,31 @@ func (s *Service) computeTps(block *types.Block) uint64 {
 			parent, err := fullBackend.BlockByNumber(context.Background(), rpc.BlockNumber(block.NumberU64()-1))
 			if err != nil {
 				log.Error("error getting parent block %s: %s", block.ParentHash().String(), err.Error())
-				return c_uintMaxValue
+				return c_statsErrorValue
 			}
-			parentTime = parent.Time()
-			parentTps = c_uintMaxValue
+			parentTime = int64(parent.Time())
+			parentTps = c_statsErrorValue
 		} else {
 			log.Error("not running fullnode, cannot get parent block")
-			return c_uintMaxValue
+			return c_statsErrorValue
 		}
 	}
 
-	instantTps := uint64(len(block.Transactions()))
-	if dt := block.Time() - parentTime; dt != 0 { // this is a hack to avoid dividing by 0
+	instantTps := int64(len(block.Transactions()))
+	if dt := int64(block.Time()) - parentTime; dt != 0 { // this is a hack to avoid dividing by 0
```

Comment:
I also don't love casting away from times so much but that's more for your backend to interpret. So if it works as is then it's fine.